### ISSUE TITLE
Wrong symbolic solution of quadratic equation

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#9575]
Consider the following MWE, coming from a valve model that uses a DP defined for generic flow models, positive with pumps, but negative with valve:
```modelica
model M
  Real Cv;
  input Real Q(start = 1);
  input Real DP(start = -1e5);
equation
  DP*Cv^2.0 = (-1e7)*Q^2.0;
 end M;
```
It fails with 
```
Model error: Argument of sqrt(-10000000.0) was -1e+07 should be >= 0
```
This happens because the quadratic equation is solved for Cv as:
```
(assign) Cv := $_signNoNull($TMP$VAR$1$0X$ABS) * abs((-10000000.0) ^ 0.5 * abs(Q) * (1.0 / DP) ^ 0.5)
```
which is wrong because `(a/b)^0.5` is well defined when `a < 0` and `b < 0`, while `a^0.5/b^0.5` is not. The square root terms should be kept together.

# Expected result
The symbolic solution should be
```
(assign) Cv := $_signNoNull($TMP$VAR$1$0X$ABS) * abs((-10000000.0/DP) ^ 0.5 * abs(Q))
```
Keeping @pepmts, @AndreaBartolini in the loop.

**Comment by phannebohm** [comment]
The old frontend simplify had a rewrite where it distributed fractional exponents even for negative factors, which is indeed not valid in the case of your MWE. I fixed that in #9619.

**Comment by casella** [comment]
Thanks @phannebohm!

> The old frontend simplify had a rewrite where it distributed fractional exponents even for negative factors, which is indeed not valid in the case of your MWE. I fixed that in #9619.

Why is the old frontend still relevant?

**Comment by perost** [comment]
> Thanks @phannebohm!
> 
> > The old frontend simplify had a rewrite where it distributed fractional exponents even for negative factors, which is indeed not valid in the case of your MWE. I fixed that in #9619.
> 
> Why is the old frontend still relevant?

The backend still uses the old frontend structures, so to simplify expressions it has to use the expression simplifier from the old frontend.
